haystack/components/websearch/searchapi.py

```diff
@@ -101,12 +101,12 @@ def run(self, query: str) -> Dict[str, Union[List[Document], List[str]]]:
         :raises SearchApiError: If an error occurs while querying the SearchApi API.
         """
         query_prepend = "OR ".join(f"site:{domain} " for domain in self.allowed_domains) if self.allowed_domains else ""
-
-        payload = json.dumps({"q": query_prepend + " " + query, **self.search_params})
-        headers = {"Authorization": f"Bearer {self.api_key.resolve_value()}", "X-SearchApi-Source": "Haystack"}
```

Comment:
Passing `headers` should work properly, according to the [docs](https://www.searchapi.io/docs/google#api-parameters-api-key-api-key).
Why change that?